src/hazelcore/semantics/OpSeq.re

```diff
@@ -70,3 +70,26 @@ let make_inconsistent =
     let set_skel = Skel.BinOp(InHole(TypeInconsistent, u), op, skel1, skel2);
     (OpSeq(set_skel, seq), u_gen);
   };
+
+let rec is_complete_skel =
```

Comment:
completeness according to POPL 17 paper actually requires no type holes, but for renumbering we don't want to do that. can you add an optional bool argument to all the is_complete functions called "check_type_holes" that serves as a flag for whether to count type holes or not